### DISCUSSION TITLE
<title>Two-factor authentication failed.

### DISCUSSION THREAD

**Discussion by xmcs1** [community/community#124450] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

My e-mail is xmji12007@163.com,I have a password and username and use 2fa---Authy,And it works fine, but I can't log into my github account. After entering the 2fa code it prompts:Two-factor authentication failed.Please help me.I need my account,Thank you very much

**Comment by CopperEagle** [comment]
Hello @xmcs1 

Could you tell us which option for 2FA you use? An authenticator app? The mobile option? 

Also make sure you have your recovery codes ready **in case** you need to reconfigure 2FA. Beware, though, that each code can only be used once.

**Reply by xmcs1** [reply]
Authy iphone13

**Comment by Soumya-RanjanSwain** [comment]
if you would like to enable 2fa, go to settings then password and authentication, there you will see a 2factor authentication option, form there you can select any desired method like mobile authenticator application like google authenticator, Microsoft authenticator etc., or sms/text message or security keys or github mobile.

and make sure to download the recovery codes, so that in case you forget you can still log in and can change authentication methods.

if it answers your question, you can mark this answered

**Reply by xmcs1** [reply]
I'm sorry. You're not answering the question.

**Reply by Soumya-RanjanSwain** [reply]
Oh I got it, you already used 2FA, Don't you have any other authentication method?
How are you using your account if you don't have the valid 2FA code

**Reply by xmcs1** [reply]
There is no other way now. authy works normally, but I can't log in after filling in the 6-digit code.

**Comment by xmcs1** [comment]
After such a long time has passed, I wonder if anyone can offer a solution?

**Comment by Manuch-Mojgani** [comment]
I have another issue like that and I tried any way like opening discussion and trying to contact github support but no ones answer:

My github account is logged out and when I try to login to my github account it sends an sms to my number for 2FA but give me 500 internal error said "We track these errors automatically, but if the problem persists feel free to contact us. In the meantime, try refreshing."
It also happens when I try to reset password. it sends me the sms but got the same 500 internal page error too.
I really need help cause I can not login to my personal github account right now and it has been a while.
what should I do know?

https://github.com/orgs/community/discussions/156093

and

https://github.com/orgs/community/discussions/156262
